### DISCUSSION TITLE
Shell aliases

### DISCUSSION THREAD

**Discussion by diegoherranz** [zephyrproject-rtos/zephyr#69156] [Q&A] [answered]
Hi,

Is it possible to define shell aliases?
I'd like to use the GPIO shell and being able to use custom-defined names rather than the GPIO device and number.

That is, I would like to be able to run:
`gpio set relay2 1`
instead of:
`gpio set gpio@40020800 7 1 ` (pin 7 of gpioc at address 40020800)

I had a look but couldn't find anything and it would be a very useful feature.

Thanks!

**Comment by nixward** [comment, marked as answer]
Hi, you can use an overlay or add this property to your gpio device tree nodes to define line names for each pin of your port like this:

https://github.com/zephyrproject-rtos/zephyr/blob/main/boards%2Farm%2Fnrf52840dk_nrf52840%2Fnrf52840dk_nrf52840.dts#L153

But at present when using the shell you still need to specify the port before the line-name. You can use tab complete to discover what line names are provided by the port once the port is specified in the shell command.

**Comment by diegoherranz** [comment]
Thanks for your reply. I had missed the `gpio-line-names` before.
As you say, it's not perfect in the sense that I'd prefer to avoid specifying the device, but it gets much closer :)
Thanks again.
